mangaki/mangaki/management/commands/compare.py

```diff
@@ -1,121 +1,75 @@
 from mangaki.utils.pca import MangakiPCA
 from mangaki.utils.zero import MangakiZero
-
 from mangaki.utils.values import rating_values
 
-import matplotlib.pyplot as plt
 
-TEST_SIZE = 0.2  # 20% is kept for the test set
+NB_SPLIT = 5  # Divide the dataset into 5 buckets
 
 
 class Experiment(object):
-    X = None
-    y = None
-    X_test = None
-    y_test = None
-    y_pred = None
-    results = {}
-    algos = None
-    def __init__(self, PIG_ID=None):
-        self.algos = [MangakiALS(20), MangakiWALS(20), MangakiSVD(20), MangakiPCA(20), MangakiKNN(20), MangakiZero()]
-        # self.results.setdefault('x_axis', []).append()
-        self.make_dataset(PIG_ID)
-        self.execute()
-
-    def clean_dataset(self):
-        self.X = []
-        self.y = []
-        self.X_test = []
-        self.y_test = []
-        self.y_pred = []
+    def __init__(self, dataset_name):
+        self.algos = [
+            lambda: MangakiALS(20),
+            # lambda: MangakiWALS(20),
+            lambda: MangakiSVD(20),
+            # lambda: MangakiPCA(20),
+            # lambda: MangakiKNN(20),
+            # lambda: MangakiZero()
+        ]
+        self.anonymized = None
+        self.load_dataset(dataset_name)
+        self.compare_models()
 
-    def make_dataset(self, PIG_ID):
-        self.clean_dataset()
-        with open(os.path.join(settings.BASE_DIR, '../data/ratings.csv')) as f:
-            ratings = [[int(line[0]), int(line[1]), line[2]] for line in csv.reader(f)]
-        ratings = np.array(ratings, dtype=np.object)
-        if PIG_ID:  # Let's focus on the PIG
-            pig_ratings = {}
-            for user_id, work_id, choice in ratings:
-                if user_id == PIG_ID:
-                    pig_ratings[work_id] = rating_values[choice]  # just choice for Movielens
-        self.nb_users = max(ratings[:, 0]) + 1
-        self.nb_works = max(ratings[:, 1]) + 1
-        with open(os.path.join(settings.BASE_DIR, '../data/works.csv')) as f:
-            self.works = [x for _, x in csv.reader(f)]
-        train, test = train_test_split(ratings, random_state=0, test_size=TEST_SIZE)
-        if PIG_ID:
-            train = ratings
-        self.X = train[:, 0:2]
-        self.y = list(map(lambda choice: rating_values[choice], train[:, 2]))  # train[:, 2] for Movielens
-        self.X_test = test[:, 0:2]
-        self.y_test = list(map(lambda choice: rating_values[choice], test[:, 2]))  # test[:, 2] for Movielens
-        if PIG_ID:
-            self.X_test = []
-            self.y_test = []
-            for work_id in range(self.nb_works):
-                self.X_test.append((PIG_ID, work_id))
-                self.y_test.append(0 if work_id not in pig_ratings else pig_ratings[work_id])
-            self.X_test = np.asarray(self.X_test)
-            self.y_test = np.asarray(self.y_test)
+    def load_dataset(self, dataset_name):
+        dataset = Dataset()
+        if dataset_name == 'movies':
```

Comment:
Why not have a dict of dataset names ←→ CSV filenames.
This way, we will be able to do more than movies! (and fallback to filename so that we can plug any custom CSV).